### PR TITLE
refine gcp/azure proxy and private cluster coverage

### PR DESCRIPTION
**Pull request by jianlinliu**
1. introduce `azure-ipi-fullyprivate-proxy` and `gcp-ipi-proxy-private` workflow
2. combine fullyprivate and proxy on azure, combine proxy and private on gcp, so that we can reduce duplicates, and save quotas  in ci coverage.

### CODE REVIEW

**Comment by jianlinliu** [comment]
Many rehearse jobs (they referenced the same install workflow) failed because quota issues, in the following jobs, we can confirm the two new workflow complete installation successfully. 

```
gcp-ipi-proxy-private:

https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/34020/rehearse-34020-periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-gcp-ipi-proxy-private-p2/1592391682706378752

https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/34020/rehearse-34020-periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-stable-e2e-gcp-ipi-proxy-private-p2/1592391683100643328

azure-ipi-fullyprivate-proxy:

https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/34020/rehearse-34020-periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-e2e-azure-ipi-fullyprivate-proxy-p2/1592391683247443968
```

So the PR is ready to review.

**Comment by jianlinliu** [comment]
@liangxia @jianli-wei @MayXuQQ @jinyunma pls help review.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/OWNERS [hunk 1]
```diff
@@ -0,0 +1,12 @@
+approvers:
+- jianlinliu
+- Amoghrd
```
Comment (MayXuQQ) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/OWNERS:3]: still add Amoghrd ?

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/provision/cucushift-installer-rehearse-azure-ipi-fullyprivate-proxy-provision-chain.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+chain:
+  as: cucushift-installer-rehearse-azure-ipi-fullyprivate-proxy-provision
+  steps:
+  - ref: azure-provision-resourcegroup
+  - ref: azure-provision-vnet
+  - chain: azure-provision-bastionhost
+  - ref: proxy-config-generate
+  - chain: ipi-conf-azure
+  - ref: ipi-conf-proxy
```
Comment (MayXuQQ) [inline ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/provision/cucushift-installer-rehearse-azure-ipi-fullyprivate-proxy-provision-chain.yaml:9]: - chain: ipi-conf-azure
  - ref: ipi-conf-proxy

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/provision/cucushift-installer-rehearse-azure-ipi-fullyprivate-proxy-provision-chain.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+chain:
+  as: cucushift-installer-rehearse-azure-ipi-fullyprivate-proxy-provision
+  steps:
+  - ref: azure-provision-resourcegroup
+  - ref: azure-provision-vnet
+  - chain: azure-provision-bastionhost
```
[the remark this answers is not in the document]
Comment (jinyunma) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/provision/cucushift-installer-rehearse-azure-ipi-fullyprivate-proxy-provision-chain.yaml:6]: we don't need this step `azure-provision-nat`, as this new workflow is fully private + disconnected + proxy scenario, and traffic from cluster to internet is through proxy.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/OWNERS (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/OWNERS:3]: I just copied the OWNER file from the existing folders to here, if you are suggesting to remove him, we can have dedicated PR to do that.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/provision/cucushift-installer-rehearse-azure-ipi-fullyprivate-proxy-provision-chain.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jianlinliu) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/provision/cucushift-installer-rehearse-azure-ipi-fullyprivate-proxy-provision-chain.yaml:6]: Yeah, @jinyunma 's understanding is right. Once `OUTBOUND_TYPE` is set to `UserDefinedRouting`, installer would not create any public LB for the cluster, all the vms will lost internet connectivity capability, to make the VMs to have the connectivity to complete the installation, users need to provide their own Routing solutions, NAT or proxy is just for that. In this case, we selected proxy on purpose without NAT.

File: ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/OWNERS (hunk 1 above)
Comment (liangxia) [reply ci-operator/step-registry/cucushift/installer/rehearse/azure/ipi/fullyprivate/proxy/OWNERS:3]: +1
small and simple PRs are easier to review / track

**Pull request merged**